File: medium/list-game/list_game.py
import sys
from math import sqrt

def prime_factors(a):
    factors = []
    p, n = 2, a
    while p <= sqrt(n):
        if not n % p:
            factors.append(p)
            n = n / p
            continue
        p += 1

    print(len(factors) + 1)


def main():
    num = int(input())
    prime_factors(num)

if __name__ == '__main__':
    main()

File: medium/list-game/ListGame.java
import java.util.*;
import java.io.*;

public class ListGame {

  public static void main(String[] args) {
    new ListGame();
  }

  public ListGame() {
    try {
      run();
    } catch (Exception e) {
      System.out.println("...");
    }
  }

  public void run() throws Exception {
    BufferedReader scan = new BufferedReader(new InputStreamReader(System.in));
    int n = Integer.parseInt(scan.readLine());
    System.out.println(primeFactors(n));
  }

  public int primeFactors(int a) {
    List factors = new ArrayList();
    int p = 2; int n = a;
    while(p <= Math.sqrt(a)) {
      if((n % p) == 0){
        factors.add(p);
        n = n / p;
        continue;
      }
      p++;
    }
    return factors.size();
  }
}
